vault/utils/utils.go

```diff
@@ -166,26 +157,31 @@ func GetAuthToken(client *api.Client, config map[string]interface{}) (string, er
 	return secret.Auth.ClientToken, nil
 }
 
-func authenticate(client *api.Client, config map[string]interface{}) (string, http.Header, map[string]interface{}, error) {
-	method := GetVaultParam(config, AuthMethod)
-	switch method {
-	case AuthMethodKubernetes:
-		return authKubernetes(client, config)
+func authKubernetes(client *api.Client, config map[string]interface{}) (*api.Secret, error) {
+	role := GetVaultParam(config, AuthKubernetesRole)
+	if role == "" {
+		return nil, ErrKubernetesRole
 	}
-	return "", nil, nil, fmt.Errorf("%s method: %s", method, ErrAuthMethodUnknown)
-}
 
-func authKubernetes(client *api.Client, config map[string]interface{}) (string, http.Header, map[string]interface{}, error) {
```

Comment:
Conveniently, `authAppRole()` below was already using the Vault API, so I was able to update the `authKubernetes()` method here to use the same basic flow.

I kept the same basic code logic from `buildAuthConfig()` for setting values from `VAULT_AUTH_KUBERNETES_ROLE`, `VAULT_AUTH_MOUNT_PATH` and `VAULT_AUTH_KUBERNETES_TOKEN_PATH` in the modified `authKubernetes()` as well to make sure behavior remains the same for default cases.